roles/cloud-bootstrap/tasks/main.yml

```diff
@@ -3,3 +3,8 @@
   include: network.yml
   tags:
     - network
+
+- name: Provision users and projects
+  include: create-users-projects.yml
+  tags:
+    - create-users-projects
```

Comment:
Actually my point looks easily resolved. Once we create the project, we can pass an auth parameter to the resources in network.yml, that specifies only the project and it will override in the auth provided the user running the play has access.